sre-recipes/recipes/recipe_1/currencies_recipe.py

```diff
@@ -0,0 +1,138 @@
+# Copyright 2020 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# -*- coding: utf-8 -*-
+"""
+This module contains the implementation of recipe 1
+"""
+
+import logging
+import subprocess
+from recipe import Recipe
+
+
+class CurrenciesRecipe(Recipe):
+    """
+    This class implements recipe 1, which purposefully
+    introduces latency into the frontend service.
+    """
+
+    @staticmethod
+    def _run_command(command):
+        """Runs the given command and returns any output and error"""
+        process = subprocess.Popen(command.split(), stdout=subprocess.PIPE)
+        output, error = process.communicate()
+        return output, error
+
+    @staticmethod
+    def _deploy_state(state):
+        """
+        Sets an environment variable CONVERT_CURRENCIES to given state
+        and updates the state accordingly
+        """
+        state_str = str(state).lower()
+        set_env_command = f"kubectl set env deployment/frontend CONVERT_CURRENCIES={state_str}"
+        get_pod_command = """kubectl get pod -l app=frontend -o \
+            jsonpath=\"{.items[0].metadata.name}\""""
+        logging.info('Setting env variable: %s', set_env_command)
+        logging.info('Getting pod: %s', get_pod_command)
+
+        CurrenciesRecipe._run_command(set_env_command)
+        service, error = CurrenciesRecipe._run_command(get_pod_command)
+        service = service.decode("utf-8").replace('"', '')
+        delete_pod_command = f"kubectl delete pod {service}"
+        logging.info('Deleting pod: %s', delete_pod_command)
+        CurrenciesRecipe._run_command(delete_pod_command)
+
+    def break_service(self):
+        """
+        Rolls back the working version of the given service and deploys the
+        broken version of the given service
+        """
+        print("Deploying broken service...")
+        self._deploy_state(True)
+        logging.info('Deployed broken service')
+
+    def restore_service(self):
+        """
+        Rolls back the broken version of the given service and deploys the
+        working version of the given service
+        """
+        print("Deploying working service...")
+        self._deploy_state(False)
+        logging.info('Deployed working service')
+
+    @staticmethod
+    def _get_user_input():
+        """Prompts users for an answer"""
+        answer = input('Your answer: ')
+        return answer
```

Comment:
nit: I'm not sure this needs to be a function, since it's a single line

Reply:
fixed!